File: java/trip-service-kata/src/main/java/org/craftedsw/tripservicekata/trip/TripDAO.java
package org.craftedsw.tripservicekata.trip;

import org.craftedsw.tripservicekata.exception.CollaboratorCallException;
import org.craftedsw.tripservicekata.user.User;

import java.util.List;

public class TripDAO {

    public static List<Trip> findTripsByUser(User user) {
        throw new CollaboratorCallException(
                "TripDAO should not be invoked on an unit test.");
    }

    public List<Trip> tripsBy(User user) {
        return TripDAO.findTripsByUser(user);
    }
}

File: kotlin/trip-service-kata/src/main/kotlin/org/craftedsw/tripservicekata/trip/TripDAO.kt
package org.craftedsw.tripservicekata.trip

import org.craftedsw.tripservicekata.exception.CollaboratorCallException
import org.craftedsw.tripservicekata.user.User

class TripDAO {

    companion object {
        @JvmStatic fun findTripsByUser(user: User): List<Trip> {
            throw CollaboratorCallException("TripDAO should not be invoked on an unit test.")
        }
    }

}
